src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/SpanNDHelpers.cs

```diff
@@ -0,0 +1,464 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+using Microsoft.VisualBasic;
+
+#pragma warning disable 8500 // sizeof of managed types
+
+namespace System.Numerics.Tensors
+{
+    internal static partial class SpanHelpers
+    {
+        // Replacing Unsafe.IsOpportunisticallyAligned
+        // Determines if the address is aligned at least to `alignment` bytes.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static unsafe bool IsOpportunisticallyAligned<T>(ref readonly T address, nuint alignment)
+        {
+            // `alignment` is expected to be a power of 2 in bytes.
+            // We use Unsafe.AsPointer to convert to a pointer,
+            // GC will keep alignment when moving objects (up to sizeof(void*)),
+            // otherwise alignment should be considered a hint if not pinned.
+            Debug.Assert(nuint.IsPow2(alignment));
+            return ((nuint)Unsafe.AsPointer(ref Unsafe.AsRef(in address)) & (alignment - 1)) == 0;
+        }
+
+        public static unsafe void ClearWithReferences(ref IntPtr ip, nuint pointerSizeLength)
```

Comment:
Do we really need our own custom implementation of this? Can't it just instead be based on Span / MemoryExtensions, and if we need to support greater than int.MaxValue, just partitioning it up and clearing int.MaxValue at a time?

Reply:
Its not a custom implementation. Its taken from `src\libraries\System.Private.CoreLib\src\System\SpanHelpers.cs`, its just that its internal so we don't have access to it here. Even the original implementation uses an `nuint`. If there is a better way to share/access the code I am happy to change it to that way, but our code should be 100% identical to the one in corelib.

Reply:
It's not necessarily as simple as just doing `int.MaxValue` sized clears, since we might have a `4x4` slice out of a `16x16` buffer (for example), and therefore could only clear 4 contiguous elements at a time.

But we can definitely break it down to using span for the most part, and that's really the most efficient thing to do anyways.

Reply:
Ok, refactored and removed the if and am now using spans for clear.